crates/jql-parser/src/parser.rs

```diff
@@ -33,32 +32,29 @@ use crate::{
 };
 
 /// Parses the provided input and map it to the first matching token.
-fn parse_fragment(input: &str) -> IResult<&str, Token> {
+fn parse_fragment<'a>(input: &mut &'a str) -> PResult<Token<'a>> {
```

Comment:
Forgot to call out, if you are able to parse `&[u8]` instead of `&str`, that also offers big gains